imagebuildah/stage_executor.go

```diff
@@ -1246,6 +1287,18 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 			if err != nil {
 				return "", nil, fmt.Errorf("error committing container for step %+v: %w", *step, err)
 			}
+			// Following step is just built and was not used from cache so
+			// check if --cache-to was specified if yes then attempt pushing
+			// this cache to remote repo and fail accordingly
+			if cacheKey != "" && cacheID == "" && s.executor.cacheTo != "" {
```

Comment:
`cacheID` is always empty in this block, so no need to check for it here.
`cacheKey` should never be empty if `s.executor.cacheTo` is set, so no need to check for it here.